File: 1885-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        res = 0
        mod = 1_000_000_007
        streak = 0
        for i in range(len(s)):
            if i == 0 or s[i] == s[i-1]:
                streak += 1
            else:
                streak = 1
            res = (res + streak) % mod
        return res
        

File: 1885-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.java
class Solution {
    public int countHomogenous(String s) {
        int res = 0;
        int mod = 1_000_000_007;
        int streak = 0;
        for(int i=0; i<s.length(); i++){
            if(i==0 || s.charAt(i) == s.charAt(i-1)){
                streak++;
            }
            else{
                streak = 1;
            }
            res = (res + streak) % mod;
        }
        return res;
    }
}
